### PR TITLE
Update two references to non-existent vars 

### PR DESCRIPTION
**Pull request by L1ghtmann**
What does this implement/fix? Explain your changes.
---------------------------------------------------
- Updates `legacy.mk` to no longer attempt to assign FW_PACKAGE_FILENAME to THEOS_PACKAGE_FILENAME, as the latter no longer exists.
  - Will now check for FW_PACKAGE_FILENAME's existence and warn about its deprecation and Theos' automatic handling of the package filename.
- Updates references to SDKTARGET in Windows' `iphone.mk` to _THEOS_TARGET_SDK_BIN_PREFIX.
- Adds previously missing _THEOS_TARGET_USE_CLANG_TARGET_FLAG to Windows' `iphone.mk`.

Does this close any currently open issues?
------------------------------------------
N/A

Any relevant logs, error output, etc?
-------------------------------------
N/A

Any other comments?
-------------------
The Windows `iphone.mk` changes are just for completeness since support for that platform is currently considered deprecated. Said changes have not been tested, but it now lines up with Linux's `iphone.mk`.

Where has this been tested?
---------------------------
**Operating System:** …

Linux (WSL)

**Platform:** …

**Target Platform:** …

**Toolchain Version:** …

**SDK Version:** …

### CODE REVIEW

File: makefiles/legacy.mk [hunk 1]
```diff
@@ -44,9 +44,12 @@ FW_PACKAGE_NAME = $(THEOS_PACKAGE_NAME)$(warning FW_PACKAGE_NAME is deprecated.
 FW_PACKAGE_ARCH = $(THEOS_PACKAGE_ARCH)$(warning FW_PACKAGE_ARCH is deprecated. Please migrate to THEOS_PACKAGE_ARCH.)
 FW_PACKAGE_VERSION = $(THEOS_PACKAGE_BASE_VERSION)$(warning FW_PACKAGE_VERSION is deprecated. Please migrate to THEOS_PACKAGE_BASE_VERSION.)
 FW_PACKAGE_DEBVERSION = $(THEOS_PACKAGE_VERSION)$(warning FW_PACKAGE_DEBVERSION is deprecated. Please migrate to THEOS_PACKAGE_VERSION.)
-FW_PACKAGE_FILENAME = $(THEOS_PACKAGE_FILENAME)$(warning FW_PACKAGE_FILENAME is deprecated. Please migrate to THEOS_PACKAGE_FILENAME.)
 FW_PACKAGE_STAGING_DIR = $(THEOS_STAGING_DIR)$(warning FW_PACKAGE_STAGING_DIR is deprecated. Please migrate to THEOS_STAGING_DIR.)
 
+ifneq ($(FW_PACKAGE_FILENAME),)
```
Comment (kabiroberai) [inline makefiles/legacy.mk:49]: I believe the previous intention of this code was to warn when something _reads_ `FW_PACKAGE_FILENAME` whereas this now warns when the user _writes_ that variable. Consider keeping the old code instead, but changing it to something like
```make
FW_PACKAGE_FILENAME = $(error FW_PACKAGE_FILENAME is deprecated. Please migrate to THEOS_PACKAGE_FILENAME.)
```

Comment (L1ghtmann) [reply makefiles/legacy.mk]: Oops! Thanks for catching that. Adjusted accordingly.

**Review by corywhirl06 — COMMENTED** [review]
Better

**Review by uroboro — APPROVED** [review]

**Review by leptos-null — APPROVED** [review]
Looks good to me.

Tested with

```make
after-all::
	echo $(FW_PACKAGE_FILENAME)
```

result:

```
*** FW_PACKAGE_FILENAME is deprecated. The filename for the most recent package can be found in '.theos/last_package'..  Stop.
```

**Pull request merged**